### ISSUE TITLE
[3.57] [MC 1.12.2] Redstone signal doesn't stop Gasifier from maintaining heat and using up fuel

### ISSUE THREAD

**Issue by Targren** [GlobbyPotato/Rockhounding_Chemistry#154]
Currently playing "Multiblock Madness" modpack (1.0.4) and I woke up this morning to find out that I had burned through my entire stash of syngas for some reason. After some investigation, I found that  I ran out of cracked coal, so there was nothing to process, but the gasifier was maintaining its heat level anyway. 

Fast forward to me messing around a bit and discovering redstone control, and I set up a buffer fluid tank for coal slurry to emit a redstone signal when it was less than 20%, and directed that signal to the checkered bar on the Gasifier.  The UI changed correctly (replacing the activation switch with the redstone torch icon), but even when the signal was active, the machine continued to burn fuel to maintain heat.  In case there was something wrong with my redstone design, I set up a lever instead. Same effect: the icon changes, the temperature stays up until the tank is empty. 

When I remove the redstone and turn off the machine with the switch, it works correctly. Temperature immediately starts dropping.

**Comment by GlobbyPotato** [comment]
I think when the redstone went OFF, in the cistern there were still traces of slurry so the machine recognized the recipe and mantained the heating waiting for more slurry. Probably the Gasifier was still manually ON when the signal stopped.
Before overriding the activation you need to make sure the manual Activation button is always turned OFF, because once it stops overriding, the manual control will return in command.

**Comment by Targren** [comment]
> I think when the redstone went OFF, in the cistern there were still traces of slurry so the machine recognized the recipe and mantained the heating waiting for more slurry. Probably the Gasifier was still manually ON when the signal stopped.
> Before overriding the activation you need to make sure the manual Activation button is always turned OFF, because once it stops overriding, the manual control will return in command.

Yeah, there was a little bit less, which is why I wanted to turn off the machine with redstone, so that it could wait until there was enough to actually process.  

I'm confused about the setup you describe.  Maybe I'm misunderstanding how the redstone signal affects the machines - I thought it just stopped them. So if the button is off with no redstone signal, and then applying a redstone signal stops the machine, then it would never run. 

Or does redstone signal just invert whatever the switch is manually set to? But if that was the case, it would still turn off with the glimpse of slurry in the cistern, like turning off the switch does. 

I'm very confused. :)

**Comment by GlobbyPotato** [comment]
> Yeah, there was a little bit less, which is why I wanted to turn off the machine with redstone, so that it could wait until there was enough to actually process.

The cistern on top is part of the machine already, once the slurry gets there the machine attempts to craft because of the preheating stage so it always waits for more slurry and keeps the high temp. It shouldn't be considered because it will always run, so the redstone needs to be set on an external tank. 

> I'm confused about the setup you describe. Maybe I'm misunderstanding how the redstone signal affects the machines - I thought it just stopped them. So if the button is off with no redstone signal, and then applying a redstone signal stops the machine, then it would never run.
Or does redstone signal just invert whatever the switch is manually set to? But if that was the case, it would still turn off with the glimpse of slurry in the cistern, like turning off the switch does.

The redstone signal ON/OFF cycle simply says:
- Ok mate, I got this redstone signal from the tank I'm observing for you. I'm forcing the machine ON myself whatever state the machine is now (torch button appears).
- Ok mate, I lost that signal, tank is low apparently and my guard is over for now. The machine will return back to wathever state you had manually set previously (either red or green led appears again).

The OFF state is common between redstone and manual so you always need to make sure the starting state is OFF and let the redstone do the cycle itself. In this way the machine will stop when getting in control again.
